src/ci.js

```diff
@@ -1,30 +1,56 @@
 const { v4: uuidv4 } = require('uuid')
-const { name, version } = require('.././package.json')
+const { name: collector, version } = require('.././package.json')
 
 class CI {
+  // the analytics env are more specific than the automatic ci platform env.
+  // If they've been specified we'll assume the user wants to use that value instead.
   env() {
     return({
-      "version": version,
-      "collector": `js-${name}`,      
-      ...this.ci_env()
+      ...this.ci_env(),
+      ...this.analytics_env()
```

Comment:
I think these should be the other way around, the `analytics_env` as the default, overridden with `ci_env`.

Reply:
The values in `analytics_env` are intended to facilitate overriding of the defaults inherited from `ci_env`, rather than defaults. Under most operating conditions the override variables do not need to be set.